### PR TITLE
node_go_simple go and stop code without classes

### PR DESCRIPTION
**Pull request by P4B5**
code for stop&go exercise, implemented without classes

### CODE REVIEW

File: go_and_stop/src/node_go_simple.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		velocity_msg.linear.x = 0.2;
+		velocity_msg.linear.y = 0;
+		velocity_msg.linear.z = 0;
+
+		velocity_msg.angular.x = 0;
+		velocity_msg.angular.y = 0;
+		velocity_msg.angular.z = 0;
+
+		//condition to stop the robot in case a bumper is pressed
+		if (bumper_state == true) {
```
Comment (fmrico) [inline go_and_stop/src/node_go_simple.cpp]: `if (bumper_state) {`

File: go_and_stop/src/node_go_simple.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	//subscriber
+	ros::Subscriber bumper_sub = n.subscribe("/mobile_base/events/bumper", 1, bumperCallback);
+
+	ros::Rate loop_rate(5);
+
+	int count = 0;
+
+	while (ros::ok())
+	{
+		geometry_msgs::Twist velocity_msg;
```
Comment (fmrico) [inline go_and_stop/src/node_go_simple.cpp]: use two spaces for indentation

File: go_and_stop/src/node_go_simple.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		velocity_msg.angular.y = 0;
+		velocity_msg.angular.z = 0;
+
+		//condition to stop the robot in case a bumper is pressed
+		if (bumper_state == true) {
+			velocity_msg.linear.x = 0;
+			ROS_WARN("bumper state: %d", bumper_msg.state);
+		}
+
+		ROS_INFO("linear vel: %f", velocity_msg.linear.x);
```
Comment (fmrico) [inline go_and_stop/src/node_go_simple.cpp]: Correct indentation

File: go_and_stop/src/node_go_simple.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+#include "kobuki_msgs/BumperEvent.h"  //bumper -> subscriptor
+
+
+bool bumper_state = false; //global variable for bumper_state
+
+void bumperCallback(const kobuki_msgs::BumperEvent::ConstPtr& bumper_msg)
+{
+	ROS_WARN("bumper state: %d", bumper_msg->state);
+
+	if (bumper_msg->state == 1) {
```
Comment (fmrico) [inline go_and_stop/src/node_go_simple.cpp]: `bumper_state = bumper_msg->state == kobuki_msgs::BumperEvent::PRESSED;`

File: go_and_stop/src/node_go_simple.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ros/ros.h"
+#include "kobuki_msgs/Sound.h"
+#include "geometry_msgs/Twist.h" 	  //velocity -> publicador
+#include "kobuki_msgs/BumperEvent.h"  //bumper -> subscriptor
+
+
+bool bumper_state = false; //global variable for bumper_state
```
Comment (fmrico) [inline go_and_stop/src/node_go_simple.cpp]: this name is not very intuitive. Use `is_bumper_pressed`, for example. `bumper_state` does not contains what does true value means.

**Comment by P4B5** [comment]
code revised, new pull request are coming.
PS: for indentation I'm using **4 spaces** with **hard** tab type

**Comment by fmrico** [comment]
Do not close this PR. Simply, push another commit with the source revised!!!

**Comment by P4B5** [comment]
Ok

**Comment by P4B5** [comment]
Please confirm that the changes are correct and if you can merge with master. thanks!

**Comment by fmrico** [comment]
LGTM

**Comment by fmrico** [comment]
Maybe it's better than someone else accepts the PR

**Pull request merged**